core/src/main/java/org/geysermc/geyser/util/EntityUtils.java

```diff
@@ -209,6 +210,18 @@ public static void updateMountOffset(Entity passenger, Entity mount, boolean rid
                         zOffset = displayTranslation.getZ();
                     }
                 }
+                case PLAYER -> {
+                    if (passenger instanceof  TextDisplayEntity textDisplay) {
+                        Vector3f displayTranslation = textDisplay.getTranslation();
+                        if (displayTranslation != null && textDisplay.getText() != null) {
+                            int lines = PlainTextComponentSerializer.plainText().serialize(textDisplay.getText()).split("\n").length;
+                            float multiplier = (float) Math.max(0.22f, 0.45f - (0.06f * Math.floor((lines - 4) / 2f)));
```

Comment:
Can we get some info on what all these numbers are?

Reply:
I did some tests with multiple lines, then I gave an ai those tests and it gave me a function that follows a pattern based on the number of lines

Reply:
Can we get this info verified?
And I think that it'd also be nice that we assert the behaviour by adding some tests for it